src/redux/sagas/index.js

```diff
@@ -1,12 +1,14 @@
 import { delay } from 'redux-saga';
 import { types as playerAStates } from '../../game-engine/application-states/ApplicationStatesPlayerA';
-import { call, put, takeEvery, select } from 'redux-saga/effects';
+import { call, put, takeEvery, select, all } from 'redux-saga/effects';
 import { authProvider, reduxSagaFirebase } from '../../gateways/firebase';
-import { types, loginSuccess, messageReceived, eventReceived } from '../actions';
+import { types, loginSuccess, messageReceived, eventReceived, messageSent } from '../actions';
 
 function* messageSender() {
   let state = yield select();
-  if (state.message) {
+  if (state.alwaysSend) {
+    yield delay(2000);  // for dev purposes
```

Comment:
why are all of these yield statements? I'm not very familiar with this pattern, but it seems like this would require calling `next()` four times on the generator function to get to the end of this function? I'm probably just not understanding, but maybe you could add a comment to explain?

Reply:
This is the redux-saga [effects](https://redux-saga.js.org/docs/basics/Effect.html) pattern. Every effect is yielded to the code that runs the saga, which then manages how these effects are performed. I think this makes things like cancelling actions (and all their dependent actions) easy.